File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    // 1 2 3 1 이 되면 햄버거 완성
    let cnt = 0;
    for(let i = 0; i < ingredient.length; i++){
        if(ingredient.slice(i, i+4).join('') === '1231'){
            cnt++;
            ingredient.splice(i, 4);
            i -= 3;
        }
    }
    return cnt;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    hamberger = []
    cnt = 0
    for i in ingredient:
        hamberger.append(i)
        if hamberger[-4:] == [1, 2, 3, 1]:
            cnt += 1
            for i in range(4):
                hamberger.pop()
    return cnt
    